src/com/owncloud/android/files/services/FileUploadService.java

```diff
@@ -0,0 +1,1359 @@
+     *         uploads.
+     */
+    private static boolean chunkedUploadIsSupported(OwnCloudVersion version) {
+        return (version != null && version.compareTo(OwnCloudVersion.owncloud_v4_5) >= 0);
+    }
+
+    /**
+     * Service initialization
+     */
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        Log_OC.d(TAG, "mPendingUploads size:" + mPendingUploads.size() + " - onCreate");
+        mNotificationManager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
+        mBinder = new FileUploaderBinder();
+        mDb = UploadDbHandler.getInstance(this.getBaseContext());
+        mDb.recreateDb(); //for testing only
+        
+        //when this service starts there is no upload in progress. if db says so, app probably crashed before.
+        mDb.setAllCurrentToUploadLater();
+        
+        HandlerThread thread = new HandlerThread("FileUploadService-Requester");
+        thread.start();
+        
+        mServiceLooper = thread.getLooper();
+        mServiceHandler = new ServiceHandler(mServiceLooper);
+
+        uploadExecutor = Executors.newFixedThreadPool(1);
+
+        Log_OC.d(TAG, "FileUploadService.retry() called by onCreate()");
+        FileUploadService.retry(getApplicationContext());
+    }
+    
+    @Override
+    public void onDestroy() {
+        Log_OC.d(TAG, "mPendingUploads size:" + mPendingUploads.size() + " - onDestroy");
+        mServiceLooper.quit();
+        uploadExecutor.shutdown();
+    }
+
+    private final class ServiceHandler extends Handler {
+        public ServiceHandler(Looper looper) {
+            super(looper);
+        }
+
+        @Override
+        public void handleMessage(Message msg) {
+            onHandleIntent((Intent) msg.obj, (int)msg.arg1);
+        }
+    }
+
+    @Override
+    public void onStart(Intent intent, int intentStartId) {
+        Message msg = mServiceHandler.obtainMessage();
+        msg.arg1 = intentStartId;
+        msg.obj = intent;
+        mServiceHandler.sendMessage(msg);
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        onStart(intent, startId);
+        return START_STICKY; // if service is killed by OS before calling
+                             // stopSelf(), tell OS to restart service with
+                             // null-intent. 
+    }
+
+    /**
+     * The IntentService calls this method from the default worker thread with
+     * the intent that started the service. When this method returns,
+     * IntentService stops the service, as appropriate.
+     * 
+     * Note: We use an IntentService here. It does not provide simultaneous
+     * requests, but instead internally queues them and gets them to this
+     * onHandleIntent method one after another. This makes implementation less
+     * error-prone but prevents files to be added to list while another upload
+     * is active. If everything else works fine, fixing this should be a TODO.
```

Comment:
Maybe it would be easiest to restore the FileUploader and let it do the actual uploading but this FileUploadService maintains the upload list and passes uploads request forward to FileUploader (as often as needed).